===================================================================
RCS file: /cvs/src/src/newlib/libc/string/strchrnul.c,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- src/newlib/libc/string/strchrnul.c	2011/03/28 14:52:17	1.1
+++ src/newlib/libc/string/strchrnul.c	2011/04/15 20:26:05	1.2
@@ -43,7 +43,5 @@
 {
   char *s = strchr(s1, i);
 
-  if (*s != NULL)
-    return s;
-  return (char *)s1 + strlen(s1);
+  return s ? s : (char *)s1 + strlen(s1);
 }
